app/Jobs/TileSingleImage.php

```diff
@@ -125,21 +126,22 @@ public function uploadToStorage()
     public function uploadToS3Storage($disk)
     {
         $iterator = $this->getIterator($this->tempPath);
-        $root = isset($disk->getConfig()['root']) && strlen($disk->getConfig()['root']) ?
-            $disk->getConfig()['root'] . "/" : "";
+        $config = $disk->getConfig();
+        $prefixKey = "prefix";
+        $prefix = isset($config[$prefixKey]) && strlen($config[$prefixKey]) ?
+            $config[$prefixKey] . "/" : "";
 
-        $uploads = function ($files) use ($disk, $root) {
+        $uploads = function ($files) use ($disk, $prefix) {
             $client = $this->getClient($disk);
             $bucket = $this->getBucket($disk);
-            $tmpLength = strlen(config('image.tiles.tmp_dir')) + 1;
+            $dirLength = strlen(config('image.tiles.tmp_dir')) + 1;
 
             foreach ($files as $file) {
-                $path = substr($file, $tmpLength);
-                $prefix = $path[0] . $path[1] . '/' . $path[2] . $path[3] . "/";
+                $path = substr($file, $dirLength);
                 // @phpstan-ignore-next-line
                 yield $client->putObjectAsync([
                     'Bucket' => $bucket,
-                    'Key' => "{$root}{$prefix}{$path}",
+                    'Key' => "{$prefix}{$path}",
```

Comment:
So the former `$root` is now called `$prefix`. But don't we need the former `$prefix` here still? The `$path` is the UUID of the image but we still need the "fagment" parts before it.